soufi/tests/finders/test_centos_finder.py

```diff
@@ -38,289 +44,238 @@ def make_top_page_content(self, versions):
+    def test_find(self):
         finder = self.make_finder()
-        self.assertEqual(finder.source_dirs, finder.default_source_dirs)
-
-    def test_source_dirs_setup_optimal(self):
-        finder = self.make_finder(optimal_repos=True)
-        self.assertEqual(finder.source_dirs, finder.optimal_source_dirs)
-
-    def test_source_dirs_setup_custom(self):
-        repos = ['foo', 'bar', 'baz']
-        finder = self.make_finder(repos=repos)
-        expected_repos = [f"{r}/" for r in repos]
-        self.assertEqual(finder.source_dirs, expected_repos)
+        url = self.factory.make_url()
+        self.patch(finder, 'get_source_url').return_value = url
+        disc_source = finder.find()
+        self.assertIsInstance(disc_source, yum.YumDiscoveredSource)
+        self.assertEqual([url], disc_source.urls)
+
+    def test_initializer_finds_repos_when_called_with_no_args(self):
+        name = self.factory.make_string('name')
+        version = self.factory.make_string('version')
+        get_source_repos = self.patch(centos.CentosFinder, '_get_source_repos')
+        get_binary_repos = self.patch(centos.CentosFinder, '_get_binary_repos')
+        centos.CentosFinder(name, version, SourceType.os)
+        get_source_repos.assert_called_once_with(centos.DEFAULT_SEARCH, False)
+        get_binary_repos.assert_called_once_with(centos.DEFAULT_SEARCH, False)
 
     def test__get_dirs(self):
         finder = self.make_finder()
-        versions = ['4.5', '7.0', '7.1', '8.0']
-        versions_ = versions + ['ignored']
-        data = self.make_top_page_content(versions_)
-        get = self.patch(requests, 'get')
-        get.return_value = self.make_response(data, requests.codes.ok)
-        dirs = finder._get_dirs()
-        self.assertEqual(versions, dirs)
-        get.assert_called_once_with(
-            "https://vault.centos.org/centos/", timeout=30
-        )
-
-    def test__get_path(self):
-        finder = self.make_finder()
-        data = self.make_index_page_content(finder.name, finder.version)
-        ref = finder._get_path(data)
-        self.assertEqual(f'{finder.name}-{finder.version}.src.rpm', ref)
-
-    def test__get_path_returns_None_if_not_found(self):
-        finder = self.make_finder()
-        data = self.make_index_page_content(
-            self.factory.make_string(), self.factory.make_string()
-        )
+        top_repos = ('1.0.123', '2.1.3456', 'bogus', '3.7.89', '3')
+        top_data = self.make_top_page_content(top_repos)
         get = self.patch(requests, 'get')
-        get.return_value = self.make_response(data, requests.codes.ok)
-        dir_ = self.factory.make_string('8.0') + '/'
-        self.assertIsNone(finder._get_path(dir_))
+        get.return_value = self.make_response(top_data, requests.codes.ok)
+        result = list(finder._get_dirs())
+        # Ensure that only the items we're interested in come back
+        self.assertEqual([top_repos[3], top_repos[1], top_repos[0]], result)
```

Comment:
This is incredibly unreadable. What about
`self.assertThat(result, SameMembers(['1.0.123', '2.1.3456', '3.7.89']))`

Reply:
That's fair.  The intent was to demonstrate that the results were coming back in a certain order, and skipping the bogus one.  Something clearly went wrong with the execution.